pkg/scheduler/framework/framework_test.go

```diff
@@ -1935,3 +1972,343 @@ func TestUpdatePolicySnapshotStatusFrom(t *testing.T) {
 		})
 	}
 }
+
+// TestNewSchedulingDecisionsFrom tests the newSchedulingDecisionsFrom function.
+func TestNewSchedulingDecisionsFrom(t *testing.T) {
+	topologySpreadScore1 := int32(1)
+	affinityScore1 := int32(10)
+	topologySpreadScore2 := int32(0)
+	affinityScore2 := int32(20)
+
+	filteredStatus := NewNonErrorStatus(ClusterUnschedulable, dummyPlugin, dummyReasons...)
+
+	testCases := []struct {
+		name                              string
+		maxUnselectedClusterDecisionCount int
```

Comment:
Hi Ryan! 

`clusterDecisonArrayLengthLimitInAPI` is a constant which corresponds to the 1000 size limit we sent in the API; its purpose is to make sure that when there are too many picked clusters, status update will not lead to a write failure.

`maxUnselectedClusterDecisionCount` is an option set on the scheduler; the default value is 20. The purpose is, as we sync'd before, to make sure that when there are less than X picked clusters, we will add some decisions about unpicked clusters to better explain to user the decision making process. If the number of picked clusters is already large enough, no information about filtered clusters will be added, so as to avoid clotting the status.

Reply:
Discussed offline; will add fixed number of unselected cluster decisions to the status.